cpp/src/arrow/type_fwd.h

```diff
@@ -525,19 +525,19 @@ std::shared_ptr<DataType> decimal256(int32_t precision, int32_t scale);
 
 /// \brief Create a ListType instance from its child Field type
 ARROW_EXPORT
-std::shared_ptr<DataType> list(const std::shared_ptr<Field>& value_type);
+std::shared_ptr<DataType> list(std::shared_ptr<Field> value_type);
```

Comment:
@bkietz @pitrou should we treat this as a breaking change? If merging this causes no issue on the next release, I'm going to be more aggressive refactoring signatures moving forward (making sure I don't introduce bugs causes by `std::move`).

Reply:
We've already made such changes in the past and nobody complained. This doesn't morally change the API (any existing usage should still work properly) and we don't make any promises about the ABI.